### ISSUE TITLE
CUB200 cannot match any papers' result, although cars196 performs better.

### ISSUE THREAD

**Issue by unluckydan** [ronekko/deep_metric_learning#3]
Do you have any idea about this? And I notice one thing, your crop is 224, but these four papers use 227. So I have no idea about what kinds of thing make the difference in CUB200.

**Comment by ronekko** [comment]
Thank you for sharing your results.
And I also observed similar results.

One of the purpose of this repository is to compare the performances among various methods in the same condition, rather than to exactly reproduce their individual results.
So there are some difference of configurations and results between the papers and this repository's.

In particular, for all methods the experimental configuration is roughly according to the paper of N-pair loss, i.e. the pre-trained network is BVLC's GoogLeNet and [its input size is 224](https://github.com/BVLC/caffe/blob/master/models/bvlc_googlenet/train_val.prototxt#L12).

I'm wondering that overall results reported in [2] are significantly lower than other papers.
I suspect one of the causes comes from that they used GoogLeNet with batch normalization (in contrast to BVLC's one does not use BN).

**Comment by unluckydan** [comment]
I will try to figure it out, if there is any news I will let you know.

**Comment by ronekko** [comment]
That's grateful. Thank you!

**Comment by unluckydan** [comment]
I modified something like dropout and using 227 cropped images, but nothing improves. That is very wired.

**Comment by qk-huang** [comment]
Hello, can you share your results? Thanks

**Comment by huanhuancao** [comment]
> Hello, can you share your results? Thanks

So do you get good performs on the three datasets?  I got these [best] soft results by main_n_pair_mc.py.
- cub200_2011: 48.244   - Online_products: 55.925    - CARS196: 40.290
